Code/Framework/AzCore/AzCore/StringFunc/StringFunc.cpp

```diff
@@ -2320,6 +2321,26 @@ namespace AZ::StringFunc
             }
             return inout;
         }
+
+        void MakeUniqueFilenameWithSuffix(
+            const AZStd::string& directoryPath, const AZStd::string& filename, AZStd::string& outFilePath, const AZStd::string& suffix)
+        {
+            ConstructFull(directoryPath.c_str(), filename.c_str(), outFilePath);
+            if (AZ::IO::FileIOBase::GetInstance()->Exists(outFilePath.c_str()))
+            {
+                AZ::IO::Path oldPath = outFilePath;
+                AZ::IO::PathView extension = oldPath.Extension();
+                AZ::IO::PathView stem = oldPath.Stem();
+                int value = 1;
+                AZStd::string originalFname;
+                do
+                {
+                    originalFname = AZStd::string(stem.Native()) + suffix + AZStd::to_string(value++);
```

Comment:
the only problem with this methodology is if there's an existing file that's been through this on a previous iteration, such as "myFile1.ext" which will become "myFile11.ext" here, instead of "myFile2.ext". I don't think that's a big deal though, and it prevents hilarity where names like "theBest1.me" would become "theBest2.me", so I guess I'm fine with it :P

Reply:
That is why I added the option of a suffix so you could have -new to give the "theBest1-new1.me". Could add an option to look for trailing numbers if it was thought necessary😀

Reply:
ahh, very good :)